### PR TITLE
feature: better provisioning process

### PR DESCRIPTION
**Pull request by ireznice**

### CODE REVIEW

File: lib/online_kitchen/labmanager4.rb [hunk 1]
```diff
@@ -9,11 +9,11 @@
 require 'active_support/core_ext/object/blank'
 
 module OnlineKitchen
-  class DeployError < StandardError
-  end
+  # class DeployError < StandardError
```
Comment (petrSchreiber) [inline lib/online_kitchen/labmanager4.rb]: Please remove commented section

File: lib/online_kitchen.rb [hunk 2]
```diff
@@ -79,6 +79,8 @@ def setup_metriks
 
   require 'online_kitchen/labmanager'
   require 'online_kitchen/labmanager4'
-  require 'online_kitchen/workers/lab_manager_provision'
+  # require 'online_kitchen/workers/lab_manager_provision'
```
Comment (petrSchreiber) [inline lib/online_kitchen.rb]: Please remove commented section

File: lib/online_kitchen/labmanager4.rb [hunk 3]
```diff
@@ -93,62 +155,62 @@ def name
       vm[:name]
     end
 
-    def sleep_between_tries(config)
-      sleep(@rand_gen.rand(config[:try_delay_min]..config[:try_delay_max]))
-    end
-
-    def wait_for_machine_deployed(config, machine_id)
-      uri_get_machine = URI("#{config[:service_endpoint]}machines/#{machine_id}")
+#    def sleep_between_tries(config)
```
Comment (petrSchreiber) [inline lib/online_kitchen/labmanager4.rb]: Please remove commented section

File: lib/online_kitchen/labmanager4.rb [hunk 4]
```diff
@@ -93,62 +155,62 @@ def name
...
-
-      (config[:max_request_retries] / 8).ceil.times.each do
-        machine = Net::HTTP.start(uri_get_machine.host, uri_get_machine.port) do |http|
-          request = Net::HTTP::Get.new uri_get_machine
-          request.basic_auth config[:username], config[:password]
-
-          response = http.request request # Net::HTTPResponse object
-          JSON.parse(response.body)
-        end
+#    def wait_for_machine_deployed(config, machine_id)
```
Comment (petrSchreiber) [inline lib/online_kitchen/labmanager4.rb]: Please remove commented section

File: lib/online_kitchen/labmanager4.rb [hunk 5]
```diff
@@ -198,17 +268,18 @@ def provision_machine(opts = {})
 
       request_id = deploy_machine(config, opts)
       machine_id = get_machine_id(config, request_id)
-      wait_for_machine_deployed(config, machine_id)
-      start_machine(config, machine_id)
-      ips = wait_for_machine_ips(config, machine_id)
+      # wait_for_machine_deployed(config, machine_id)
```
Comment (petrSchreiber) [inline lib/online_kitchen/labmanager4.rb]: Please remove commented section

File: lib/online_kitchen/labmanager4.rb [hunk 6]
```diff
@@ -198,17 +268,18 @@ def provision_machine(opts = {})
...
-      OnlineKitchen.logger.info "Got PC with IP: #{@vm[:ip]}, name: #{@vm[:name]}"
-      self
-    rescue IPsError
-      @vm = { name: get_machine_name(machine_id, opts), ip: 'not obtained' }
-      OnlineKitchen.logger.info "Got PC with IP: #{@vm[:ip]}, name: #{@vm[:name]}"
+      # @vm = { name: get_machine_name(machine_id, opts), ip: ips.join(', ') }
+      @vm = { name: get_machine_name(machine_id, opts), ip: '' }
+      OnlineKitchen.logger.info "Got PC with name: #{@vm[:name]}"
       self
+#    rescue IPsError
```
Comment (petrSchreiber) [inline lib/online_kitchen/labmanager4.rb]: Please remove commented section

File: lib/online_kitchen/workers/lab_manager_deploy.rb [hunk 7]
```diff
@@ -0,0 +1,102 @@
+require 'benchmark'
+require 'online_kitchen/workers/lab_manager_equip'
+
+module OnlineKitchen
+  # deploys specified machine via labManager
+  class LabManagerDeploy
+    include Sidekiq::Worker
+    sidekiq_options queue: :lab_manager, retry: false
+
+    def perform(deploy_arg)
```
[the remark this answers is not in the document]
Comment (ireznice) [reply lib/online_kitchen/workers/lab_manager_deploy.rb]: gonna be done in the future feature

File: lib/online_kitchen/workers/lab_manager_equip.rb [hunk 8]
```diff
@@ -0,0 +1,78 @@
+require 'benchmark'
+require 'online_kitchen/workers/lab_manager_equip'
+
+module OnlineKitchen
+  # deploys specified machine via labManager
+  class LabManagerEquip
+    include Sidekiq::Worker
+    sidekiq_options queue: :lab_manager, retry: false
+
+    def perform(equip_arg)
```
[the remark this answers is not in the document]
Comment (ireznice) [reply lib/online_kitchen/workers/lab_manager_equip.rb]: future feature

**Pull request merged**